liquidprompt

```diff
@@ -3310,62 +3417,68 @@ _lp_default_theme_prompt_data() {
         LP_SHLVL=
     fi
 
-    if _lp_python_env_color; then
-        LP_VENV=" $lp_python_env_color"
+    if _lp_python_env_color ; then
+        LP_VENV="$lp_python_env_color"
     else
         LP_VENV=
     fi
 
     if _lp_node_env_color; then
-        LP_NODE_VENV=" $lp_node_env_color"
+        LP_NODE_VENV="$lp_node_env_color"
     else
         LP_NODE_VENV=
     fi
 
     if _lp_ruby_env_color; then
-        LP_RUBY_VENV=" $lp_ruby_env_color"
+        LP_RUBY_VENV="$lp_ruby_env_color"
     else
         LP_RUBY_VENV=
     fi
 
     if _lp_kubernetes_context_color; then
-        LP_KUBECONTEXT=" $lp_kubernetes_context_color"
+        LP_KUBECONTEXT="$lp_kubernetes_context_color"
     else
         LP_KUBECONTEXT=
     fi
 
     if _lp_terraform_env_color; then
-        LP_TFSPACE=" $lp_terraform_env_color"
+        LP_TFSPACE="$lp_terraform_env_color"
     else
         LP_TFSPACE=
     fi
 
     if _lp_container_color; then
-        LP_CONTAINER=" $lp_container_color"
+        LP_CONTAINER="$lp_container_color"
     else
         LP_CONTAINER=
     fi
 
     if _lp_software_collections_color; then
-        LP_SCLS=" $lp_software_collections_color"
+        LP_SCLS="$lp_software_collections_color"
     else
         LP_SCLS=
     fi
 
     if _lp_aws_profile_color; then
-        LP_AWS_PROFILE=" $lp_aws_profile_color"
+        LP_AWS_PROFILE="$lp_aws_profile_color"
```

Comment:
I've though about this more, and I don't think these changes are needed, or even correct.

By changing the values of `LP_AWS_PROFILE` and similar, we need to add lots of notes to the changelog and release notes, and those who want to use the "old" way will need to change their configs.

The better way to do it is on line 3277 use the values of the return variables:
```diff
-    local dev_all=("${LP_SCLS}" "${LP_AWS_PROFILE}" "${LP_CONTAINER}" "${LP_VENV}" "${LP_NODE_VENV}" "${LP_RUBY_VENV}" "${LP_TFSPACE}" "${LP_KUBECONTEXT}" "${LP_CMAKE}")
+    local dev_all=("${lp_software_collections_color-}" "${lp_aws_profile_color-}" "${lp_container_color-}" "${lp_python_env_color-}" "${lp_node_env_color-}" "${lp_ruby_env_color-}" "${lp_terraform_env_color-}" "${lp_kubernetes_context_color-}" "${lp_cmake_color-}")
```
(Probably want to wrap that line to make it readable.)

This also means we can change our code here to keep `LP_AWS_PROFILE` and similar exactly the same by adding our bracket wrapping in this location:
```bash
        LP_AWS_PROFILE=" ${LP_MARK_DEV_OPEN}${lp_aws_profile_color}${LP_MARK_DEV_CLOSE}"
```

Reply:
I would rather make the default config use brackets for open and close and space + double brackets for middle sep ("] ["). That way we get the old behavior.

I still think is a bad design, speaking ergonomics: it adds a lot of visual clutter and confusion with the path section. I'd rather have a good design and a cumbersome changelog (not so many people read it anyway) as a default.

We may also add a new theme, to get the old behavior just a command away.

Reply:
> I still think is a bad design

I agree, your new design is better. But we need to keep compatibility, which means these variables needs to stick around. I see these options:

1. Leave the variables as they are/were, but add the new `LP_DEV_ENVS` to PS1 in the default theme, and don't use `LP_AWS_PROFILE ` and similar.
2. Remove `LP_AWS_PROFILE ` and similar, but make a theme that uses them instead.

I don't see any reason to pick 2 over 1.

Reply:
Oh, I understand what you want, now, sorry for being slow. Yes, I think that’s definitely the way to go.